libs/dal/src/repositories/job/job.repository.ts

```diff
@@ -101,6 +108,7 @@ export class JobRepository extends BaseRepository<JobDBModel, JobEntity, Enforce
       type: StepTypeEnum.TRIGGER,
       _environmentId: environmentId,
       _subscriberId: subscriberId,
+      ...(digestKey && { [`payload.${digestKey}`]: digestValue }),
```

Comment:
The issue was that the jobs that should have been digested in the second digest key were retrieved here, for the first digest key. So, when the run for the second digestKey was getting the events, they were not included for him.